testsuite/tests/tsan/waitgroup_stubs.c

```diff
@@ -8,9 +8,11 @@
 #define MAX_WAITGROUP   8
 #define SPIN_WAIT_MS    10
 
+/* waitgroup inspired by Golang's `sync.WaitGroup`. This version does *not*
+ * allow to restart a waitgroup. */
 typedef struct {
-  unsigned    limit; /* Number of threads participating to the checkpoint */
-  atomic_uint count; /* Number of threads that have reach the checkpoint */
+  unsigned    limit; /* Number of threads participating in the checkpoint */
+  atomic_uint count; /* Number of threads that have reached the checkpoint */
```

Comment:
My oh my, someone has been committing comment cleanups and feature changes together! 🙀

(No need to change, I don't care.)